website/src/content/development/forms.mdx

```diff
@@ -0,0 +1,673 @@
+---
+title: Forms
+---
+
+In the previous sections we learned how to start building the actual User Interface by using UI components and fetching data. One of the most common requirements when implementing a User Interface is about forms.
+
+To facilitate the development of UI forms, we provide many different ready-to-use input and field components based on [commercetools Design System](https://github.com/commercetools/ui-kit).
+
+# Form state management
+
+Forms generally consist of a group of inputs that users interact with to eventually trigger some actions.
+In this process all user interactions need to be tracked, managed, and reflected in the UI. To achieve this, a form needs to maintain its own state.
+
+At commercetools we have settled on using [Formik](https://formik.org) as the preferred form state management library when building User Interfaces.
+
+Formik comes with several built-in features such as validations, array fields, input states, async submission and many more.
+
+At a bare minimum, implementing a form could look something like this:
+
+```jsx
+import { useFormik } from 'formik';
+import { useApplicationContext } from '@commercetools-frontend/application-shell-connectors';
+import LocalizedTextField from '@commercetools-uikit/localized-text-field';
+import LocalizedTextInput from '@commercetools-uikit/localized-text-input';
+import PrimaryButton from '@commercetools-uikit/primary-button';
+import validate from './validate';
+
+const ChannelsForm = () => {
+  const { dataLocale, languages } = useApplicationContext((context) => ({
+    dataLocale: context.dataLocale,
+    languages: context.project.languages,
+  }));
+  const formik = useFormik({
+    initialValues: {
+      // A Channel's `name`: https://docs.commercetools.com/api/projects/channels
+      name: LocalizedTextInput.createLocalizedString(languages),
```

Comment:
This was confusing to me because we're initializing form data but we're not using anything about the channels.
As this is the first example of this page and the `initialValues` is explained more in detail later, I think we maybe could omit it here.

Reply:
Hmm I can add a comment making it more explicit that it's the initial data but I would keep the code as-is because technically is correct. The point here is also to show that the props you pass to the field related to the form values.